cfep-05.md

```diff
@@ -0,0 +1,166 @@
+
+```
+conda install -c conda-forge/label/rc_matplotlib matplotlib
+```
+
+We, as the conda-forge community need to arrive at a consensus for what labels
+we will affix to packages so that there is a consistent user experience.
+
+## Backwards Compatibility
+
+There are only minimal concerns of backwards compatibility.
+The number of feedstocks that issue pre-releases is small, mitigating
+backwards compatibility issues.
+
+## Specification
+
+Mark the pre-release package on anaconda.org as "dev" or "rc" by adding them to 
+the appropriate label.
+The semantics of these labels should generally follow the
+[guidelines](https://docs.python.org/devguide/devcycle.html#stages) that Python
+itself follows.
+
+- "rc": [Beta](https://docs.python.org/devguide/devcycle.html#beta) and [Release
+  Candidate](https://docs.python.org/devguide/devcycle.html#release-candidate-rc)
+  (RC). No new features. Bugfix only.
+
+- "dev": [Pre-Alpha](https://docs.python.org/devguide/devcycle.html#pre-alpha)
+  and [Alpha](https://docs.python.org/devguide/devcycle.html#alpha). These are
+  still packages that could see substantial changes
+  between the dev version and the final release.
+
+See the Appendix for dev and rc version order specifics.
+
+To create a `dev` or `rc` package a PR can be issued into the `dev` or `rc` branch of the
+feedstock.
+This branch must change the `recipe/conda_build_config.yaml` to point to the `dev_<package_name>` or `rc_<package_name>` label.
+
+For example, matplotlib rc releases would include:
+```yaml
+...
+channel_targets:
+  - conda-forge rc_matplotlib
+```
+
+If a pre-release build of B depends on a pre-release build of A, then A should have,
+```yaml
+...
+channel_targets:
+  - conda-forge rc_A
+```
+while B should have,
+```yaml
+...
+channel_sources:
+  - conda-forge/label/rc_A,conda-forge,defaults
+channel_targets:
+  - conda-forge rc_B
+```
+in `recipe/conda_build_config.yaml` in their respective feedstocks.
+
+NOTE: A rerender needs to happen for these changes to reflect in CI files.
+
+## Alternatives
+
+* Upload pre-release packages to a different anaconda.org user, perhaps
+  `conda-forge-pre`
+* Dump all pre-release packages into one label: "pre"
+
+We understand the proposed approach does not have the same user experience
+as `pip --pre`, which seems simpler and easier to use.
+Currently this level of functionality is not possible with conda while
+maintaining the stability of environment we currently provide.
+This CFEP proposes and initial step toward supporting dev/rc releases.
+As we explore this functinality more we can re-evaluate the user and maintainer
+expreience and how best to provide this functionality.
+
+## FAQ
+
+* What about "alpha" and "beta" labels?
+
+There is not yet heavy use of labels on the conda-forge channel on anaconda.org.
+Given the light usage of labels thus far it seems rather unnecessary to
+introduce many labels.  "dev" and "rc" seem like a nice compromise.
```

Comment:
Just FYI for those who haven't seen it - there's two competing PEPs for increasing the release cadence of python. One of them (see PEP text and discussion [here](https://discuss.python.org/t/pep-605-a-rolling-feature-release-stream-for-cpython/2418)) is proposing to add a rolling release stream with alphas (ABI-breaking) and betas (stable).

The PEP is not accepted yet, and this CFEP could of course be changed later if necessary, so I don't think there's anything to do, I just wanted to point this out, because it touches on this topic.

Reply:
Yes, we could change the policy later if needed. Personally, I'd assume future CPython alphas could go into `python_dev` and betas into `python_dev` fitting nicely with this proposal.

But basically, I just wanted to say thanks for the reference! (Though, TBH, I TLDR;'d that one, haha...)

Reply:
Actually, it was the other [PEP](https://www.python.org/dev/peps/pep-0602/) that ended up being accepted. This means a smaller change compared to the status quo, just a faster release cadence.